lib/install/data/data.go

```diff
@@ -256,5 +256,12 @@ func (d *Data) CopyNonEmpty(src *Data) error {
 
 	d.Timeout = src.Timeout
 
+	if src.DNS != nil {
```

Comment:
do you need this logic? you should just check if src.DNS.IsSet, and then copy the value to d is enough.